src/Discovery.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace Http\Discovery;
+
+interface Discovery
```

Comment:
I wonder if this interface is really necessary. Since we are in static context, you can't even check the instance. So all Discovery objects are treated as final static classes. WDYT?